editor/code_editor.cpp

```diff
@@ -321,40 +328,52 @@ void FindReplaceBar::_get_search_from(int &r_line, int &r_col) {
 }
 
 void FindReplaceBar::_update_results_count() {
-	if (results_count != -1) {
+	if (!needs_to_count_results && (result_line != -1)) {
+		results_count_to_current += (flags & TextEdit::SEARCH_BACKWARDS) ? -1: 1;
+
+		if (results_count_to_current > results_count)
+			results_count_to_current = results_count_to_current - results_count;
+		else if (results_count_to_current == 0)
+			results_count_to_current = results_count;
+
 		return;
 	}
 
+	needs_to_count_results = false;
+
 	results_count = 0;
+	results_count_to_current = 0;
 
 	String searched = get_search_text();
 	if (searched.is_empty()) {
 		return;
 	}
 
-	String full_text = text_editor->get_text();
+	for (int i=0; i < text_editor->get_line_count(); i++) {
+		String line_text = text_editor->get_line(i);
```

Comment:
Wonder if we should covert this to call `text_edit->search` rather then re-implementing a version of it.

Reply:
Didn't understand? It was counting all matches, now it is countgin all matches but setting also "matches to current".